### PR TITLE
Fix: Lock can be acquired after lease duration if the current owner has been terminated unexpectedly when using skipBlockingWait=true

### PR DESCRIPTION
**Pull request by shetsa-amzn**
*Issue #, if available: https://github.com/awslabs/amazon-dynamodb-lock-client/issues/44*

*Description of changes:*
The change in this PR after resolving few conflicts 

https://github.com/awslabs/amazon-dynamodb-lock-client/pull/79

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: src/main/java/com/amazonaws/services/dynamodbv2/util/LockClientUtils.java [hunk 1]
```diff
@@ -24,12 +24,11 @@
...
-     * Calls System.nanoTime() and converts it to milliseconds. This is used by the lock client, since the lock client
-     * uses milliseconds as its base unit instead of nanoseconds.
+     * Calls System.currentTimeMillis().
      *
-     * @return the current time in milliseconds, but not since epoch. It is only useful for elapsed time, not absolute time.
+     * @return the current time in milliseconds.
      */
     public long millisecondTime() {
-        return System.nanoTime() / 1000000;
+        return System.currentTimeMillis();
```
[the remark this answers is not in the document]
Comment (qvissak) [reply src/main/java/com/amazonaws/services/dynamodbv2/util/LockClientUtils.java:32]: Isn't updating this pre-existing method outside the scope of this CR? Unless I am missing something after reading https://stackoverflow.com/questions/351565/system-currenttimemillis-vs-system-nanotime this looks like an enhancement.

[the remark this answers is not in the document]
Comment (tso) [reply src/main/java/com/amazonaws/services/dynamodbv2/util/LockClientUtils.java:32]: Updating this pre-existing method is required for the idea behind this PR, which is to use wall clock time to determine if a client can acquire the lock.

`currentTimeMillis` is wall clock time (can be converted into a time like 2023-06-13 8:37 PM)
`nanoTime` is an entirely random number that is _only_ useful to measure elapsed time

I would propose https://github.com/awslabs/amazon-dynamodb-lock-client/pull/88 is a more nuanced PR as it:
1. does not affect blocking clients by changing this method which is critical for correctness
2. surfaces clock skew to the end user as a problem they need to consider

[the remark this answers is not in the document]
Comment (qvissak) [reply src/main/java/com/amazonaws/services/dynamodbv2/util/LockClientUtils.java:32]: That makes sense. Thank you for explaining.

[the remark this answers is not in the document]
Comment (qvissak) [reply src/main/java/com/amazonaws/services/dynamodbv2/util/LockClientUtils.java:32]: I agree that we should abandon this proposed change for #88 after it is revised.

**Comment by rahulvishwanatham** [comment]
I am facing the same issue with 1.2.0 version, is this going to be merged or any other solution to this? A lock is stale in my DDB and no other instance is able to acquire the lock. I am setting withShouldSkipBlockingWait.

**Pull request closed without merging**